src/common/targetingEvaluator.ts

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { createHash } from "crypto";
+
+/**
+ * Determines if the user is part of the audience, based on the user id and the percentage range.
+ *
+ * @param userId user id from app context
+ * @param hint hint string to be included in the context id
+ * @param from percentage range start
+ * @param to percentage range end
+ * @returns true if the user is part of the audience, false otherwise
+ */
+export function isTargetedPercentile(userId: string | undefined, hint: string, from: number, to: number): boolean {
+    if (from < 0 || from > 100) {
+        throw new Error("The 'from' value must be between 0 and 100.");
+    }
+    if (to <= 0 || to > 100) {

```

Comment:
```suggestion
    if (to < 0 || to > 100) {

```

[reference](https://github.com/microsoft/FeatureManagement-Dotnet/blob/preview/src/Microsoft.FeatureManagement/Targeting/TargetingEvaluator.cs#L311) from .NET FM

This corner case is very trivial, let's just follow .NET FM

Reply:
What is the behavior if both `from` and `to` equal to 0? Should `to` be larger than `from`?

Reply:
>What is the behavior if both from and to equal to 0? 

It will never be targeted.

> Should to be larger than from?

Not necessarily

Reply:
OK, just confirm it's the design :
- `to < from`: invalid.
- `to == from`: valid but never hit.

Reply:
> OK, just confirm it's the design :
> * to < from: invalid.
> * to == from: valid but never hit.

That's correct. `from` is inclusive- which means 0 would be considered valid on that check- but `to` is explicitly exclusive- meaning 0 wouldn't be considered valid there. The .net check is: `return contextPercentage >= from && contextPercentage < to;`

Reply:
I have no problem with the results a how it’s evaluated. From the .net check you mentioned, either to==from or to<from won’t be hit. My actual question is, whether to throw an exception when from == to?

Reply:
a7f0472 : align with .NET